pkg/upgrade/upgrade.go

```diff
@@ -232,6 +233,38 @@ func UpdateFromLegacyVersion(cli client.Client, platform deploy.Platform, appNS
 	return nil
 }
 
+// CleanupDeprecatedFeatures deletes deprecated features by removing the corresponding FeatureTrackers
+func CleanupDeprecatedFeatures(ctx context.Context, cli client.Client, platform deploy.Platform, appNamespace string) error {
+	crd := &apiextv1.CustomResourceDefinition{}
+	if err := cli.Get(ctx, client.ObjectKey{Name: "featuretrackers.features.opendatahub.io"}, crd); err != nil {
+		return err
+	}
+
+	toDelete := []*featuresv1.FeatureTracker{
+		featuresv1.NewFeatureTracker("kserve-temporary-fixes", appNamespace),
+	}
+
+	var multiErr *multierror.Error
+	for _, featureTracker := range toDelete {
+		name := featureTracker.ObjectMeta.Name
+
+		if err := cli.Get(ctx, client.ObjectKey{Name: name}, featureTracker); err != nil {
+			multiErr = multierror.Append(multiErr, client.IgnoreNotFound(err))
+			continue
+		}
+
+		if err := cli.Delete(ctx, featureTracker); err != nil {
+			msg := fmt.Sprintf("error deleting FeatureTracker instance %s: %s\n", name, err)
+			fmt.Printf(msg)
```

Comment:
Don't believe it's necessary to print the error here since it looks like it gets logged in the call above in the chain if not nil.

Reply:
Agreed, deleted in the last push